docs/index.bs

```diff
@@ -3256,7 +3261,45 @@ spec: storage; urlPrefix: https://storage.spec.whatwg.org/
                   1. Let |fetchHandlerResponse| be the result of [=Create Fetch Event and Dispatch=] with |request|, |registration|, |useHighResPerformanceTimers|, |timingInfo|, |workerRealm|, |reservedClient|, |preloadResponse|, and |raceResponse|.
                   1. If |fetchHandlerResponse| is not null and not a [=network error=], and |raceFetchController| is not null, [=fetch controller/abort=] |raceFetchController|.
                   1. [=queue/Enqueue=] |fetchHandlerResponse| to |queue|.
-              1. Wait until |queue| is not empty.
+                  1. Set |fetchHandlerCompleted| to true.
+              1. Wait until |queue| is not empty or (|networkFetchCompleted| is true and |fetchHandlerCompleted| is true).
```

Comment:
@domenic Can I ask your advice on how this kinds of thing is usually written in the spec?
The explanation sounds reasonable as code, but I am afraid it is not usual way of writing spec.

Reply:
Are you concerned mainly about the parentheses? I think they are OK since they are easy to understand, but yes, it is unusual. The pattern I would use would be:

> Wait until any of the following are true:
> 
> * |queue| is not empty; or
> * |networkFetchCompleted| is true and |fetchHandlerCompleted| is true.

Reply:
Yes, I wanted to know how people usually write this because I have not seen specs using parentheses.
And, thank you for the advice.

@monica-ch Will you revise the sentences as @domenic's way?

Reply:
I'm not sure if we really really need to introduce `networkFetchCompleted` and `fetchHandlerCompleted`. For `race-network-and-fetch-handler`, the race is expressed by `queue`. Why we need these two flags?

It looks `networkFetchCompleted` doesn't wait for the promise resolve. So assuming `queue` can be still empty even when `networkFetchCompleted` is true. This will return null, which means the fetch event fallback in the handle fetch algorithm.

Also not sure why we should wait for `fetchHandlerCompleted` as well. This means we have to wait for both the network and fetch handler completion.

IIUC `race-network-and-fetch-handler` won't run the race anymore after this change.

Reply:
This change should be due to https://github.com/w3c/ServiceWorker/pull/1764#discussion_r2046691943.
I concerned that if both network and cache failed, nothing is enqueued, and waiting for queue won't finish.  I stepped back and saw the `race-network-and-fetch-handler`, and felt it may also have the same issue.
Having the way to know network and cache lookup completion should be needed to avoid the infinite queue waiting.